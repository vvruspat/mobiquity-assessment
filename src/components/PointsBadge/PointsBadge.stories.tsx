import { ComponentStory, ComponentMeta } from "@storybook/react";

import { PointsBadge } from "./PointsBadge";
import "../../index.css";

export default {
  title: "Components/PointsBadge",
  component: PointsBadge,
} as ComponentMeta<typeof PointsBadge>;

const Template: ComponentStory<typeof PointsBadge> = (args) => (
  <PointsBadge {...args} />
);

export const PointsBadgeTemplate = Template.bind({});

PointsBadgeTemplate.args = {
  points: 456,
};
