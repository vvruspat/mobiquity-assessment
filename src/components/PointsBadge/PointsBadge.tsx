import { Badge } from "../UIKit/Badge";
import { Caption } from "../UIKit/Caption";
import { Text } from "../UIKit/Text";

type PointsBadgeProps = {
  points: number;
};

export const PointsBadge = ({ points }: PointsBadgeProps) => {
  return (
    <Badge badgeStyle="info" size="l">
      <Caption>Points</Caption>
      <Text size="xxl">{points}</Text>
    </Badge>
  );
};
