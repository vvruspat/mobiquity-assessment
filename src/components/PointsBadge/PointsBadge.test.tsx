import { render, screen } from "@testing-library/react";
import { PointsBadge } from "./PointsBadge";

test("renders points badge", () => {
  render(<PointsBadge points={256} />);

  const el = screen.getByTestId("badge");
  expect(el).toHaveTextContent("Points");
  expect(el).toHaveTextContent("256");
  expect(el).toMatchSnapshot();

  const caption = screen.getByTestId("caption");
  expect(caption).toBeInTheDocument();
  expect(caption).toHaveTextContent("Points");
});
