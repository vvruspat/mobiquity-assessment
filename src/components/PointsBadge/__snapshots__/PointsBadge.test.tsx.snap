// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders points badge 1`] = `
<div
  class="badge badge_l badge_style_info badge_align_center"
  data-testid="badge"
>
  <h6
    class="caption"
    data-testid="caption"
  >
    <span
      class="text size_s regular"
      data-testid="text"
    >
      Points
    </span>
  </h6>
  <span
    class="text size_xxl regular"
    data-testid="text"
  >
    256
  </span>
</div>
`;
